philippa-choong/week7/mon/js/main.js

```diff
@@ -0,0 +1,134 @@
+// Create a Single - Page App interface to themoviedb.org which lets users search for a movie by title and shows the results on the same page, using XMLHttpRequest and jQuery.
+
+// Consider also showing the movie popularity and overview text.Bonus points for showing a movie poster thumbnail for each movie.
+
+// If a new search is performed, the old search results should be replaced(not just appended to)!
+
+// BONUS: Make each search result clickable, and use it to take the ID of the clicked movie and perform a different AJAX request to the API endpoint which gives detailed info about a movie by ID(RTFM).This lets you print out budgets, genres, cast, longer plot synopsis etc.When viewing this show page, the search results should be hidden, and when a new search is performed, the show page details should be hidden!(This will "simulate" moving between different pages, i.e.from a results page to a show page, but really you'll just be using jQuery to .show() or .hide() the "#results" or "#show" divs.)
+
+// XMLHttpRequest - 
+// to issue HTTP requests in order to exchange data between the web site and a server. 
+// built-in-method for making AJAX request (to load the contents of a URL
+// within Javascript)
+
+console.log('main.js is loaded!');
+
+
+const lookupMovie = function (movie) {
+    $('#results').html('loading....'); // clear old results and show new results next
+
+
+    const xhr = new XMLHttpRequest(); //// make a new instance, i.e. this is the constructor
+    // XMLHttpRequest() spits out the object for us. A built-in-method for making AJAX request
+
+
+    // .onload - Execute a JS immediately after a page has been loaded:
+    // this function's code will run when XHR decides the response has finished loading
+    xhr.onload = function () {
+
+        const data = JSON.parse(xhr.response);
+        // JSON - commonly used to exchange data to / from a web server.
+        // When receiving data from a web server, the data is always a string.
+        // Parse the data with JSON.parse(), and the data becomes a JavaScript object.
+        $('#results').empty() //remove previous results
+
+
+        // loop through array and find title from search 
+        const numOfResults = 10
+
+        for (let i = 0; i < numOfResults; i++) {
+            const output = data.results[i]
+            const title = output.title
+            const popularity = output.popularity
+            const overview = output.overview
+            // console.log(title, popularity, overview)
+            $('#results')
+                .append(`
+                <h2><a href="#" >${title}</a></h2> 
+                <p>popularity: ${popularity}</p>
+                <p>overview: ${overview}</p>
+                `)
```

Comment:
why not add an img here?
also why add a link if you are taking me no where? Happy with just a regular `<h2>`